packages/modules/workflow-engine-redis/src/utils/workflow-orchestrator-storage.ts

```diff
@@ -457,4 +465,97 @@ export class RedisDistributedTransactionStorage
       repeatableJobs.map((job) => this.jobQueue?.removeRepeatableByKey(job.key))
     )
   }
+
+  async #preventRaceConditionExecutionIfNecessary({
```

Comment:
I wonder when the day will come where we unify the in memory and redis logic and only use providers for the underlying storage 😄

Reply:
we need to do it def :D